tests/integ/scala/test_dataframe_writer_suite.py

```diff
@@ -152,6 +153,57 @@ def test_write_with_target_table_autoincrement(
         Utils.drop_table(session, table_name)
 
 
+def test_iceberg(session, local_testing_mode):
+    if not iceberg_supported(session, local_testing_mode):
+        pytest.skip("Test requires iceberg support.")
+
+    table_name = Utils.random_table_name()
+    df = session.create_dataframe(
+        [],
+        schema=StructType(
+            [
+                StructField("a", StringType()),
+                StructField("b", IntegerType()),
+            ]
```

Comment:
Does this work for ArrayType and MapType?

Reply:
After some testing it looks like there's several edge cases that break for structured types. This change is large already so I will address those edge cases in a second PR.